core/InstArchInfo.hpp

```diff
@@ -61,6 +62,22 @@ namespace olympia
         using TargetPipeMap = std::map<std::string, TargetPipe>;
         static const TargetPipeMap execution_pipe_map;
 
+        using TargetPipeStringMap = std::map<TargetPipe, std::string>;
+        static const TargetPipeStringMap execution_pipe_string_map;
+
+        enum class UopGenType
+        {
+            ARITH,
+            ARITH_SINGLE_DEST,
+            ARITH_WIDE_DEST,
+            NONE
```

Comment:
Since NONE is a valid `UopGenType` (e.g. instructions like ignore LMUL like certain moves and mask compares), I need to add another type called UNKNOWN or INVALID.

Reply:
Fixed.